cmd/deploy/deploy.go

```diff
@@ -26,6 +26,9 @@ func NewDeployCmd(configuration *config.Configuration) *cobra.Command {
 			cmdUtils.ExecuteParentHooks(cmd, args)
 		},
 		PersistentPostRun: func(cmd *cobra.Command, args []string) {
+			if cmd.Context().Value("dryRun") != nil && cmd.Context().Value("dryRun").(bool) {
```

Comment:
I'm not sure if it's possible to get a non-bool type for `cmd.Context().Value("dryRun")`, but as written this will panic if it's not. 

I'd do a check like this `dryRun, ok := cmd.Context().Value("dryRun").(bool)`

(If you only return one value, like this: `dryRun := cmd.Context().Value("dryRun").(bool)`, Go will panic if the value isn't a bool).